lib/Trash/TrashBackend.php

```diff
@@ -243,7 +262,8 @@ public function getTrashNodeById(IUser $user, int $fileId) {
 			$absolutePath = $this->appFolder->getMountPoint()->getMountPoint() . $path;
 			$relativePath = $trashFolder->getRelativePath($absolutePath);
 			list(, $folderId) = explode('/', $relativePath);
-			if ($this->userHasAccessToFolder($user, (int)$folderId)) {
+			// FIXME doesn't work yet since we do not have the original path
```

Comment:
@icewind1991 I could need some help here, i fail to find the connection between the appdata fileId and the entry in `oc_group_folder_trash` so I'm not sure how I could obtain the original path inside of this method.

Reply:
I guess you'll have to change the explode above in

```php
list(, $folderId, $nameAndTime) = explode('/', $relativePath);
```

then split the `$nameAndTime` (on the last `.`) and get search in the trash folder based on that.

"Going the wrong way" like this is awkward, which is why the trash api returns the `Node` instead of a `TrashItem` in the first place :)

Reply:
@icewind1991 The main issue with this is that we cannot obtain the original path just from that as the name+time are not really unique identifiers. I've pushed a commit to add the fileid to the oc_group_folders_trash table so we can properly fetch the original path to check against acl rules. Let me know what you think about that.